logs_analyzer/chapter2/scala/src/test/scala/com/databricks/apps/logs/ApacheAccessLogTest.scala

```diff
@@ -0,0 +1,54 @@
+package com.databricks.apps.logs
+
+import org.scalatest._
+
+class ApacheAccessLogTest extends FlatSpec {
+
+  "parseLogLine" should "throw IOException for null argument" in {
+    assertThrows[RuntimeException] {
+      ApacheAccessLog.parseLogLine(null)
+    }
+  }
+
+  it should "throw IOException for empty string argument" in {
+    assertThrows[RuntimeException] {
+      ApacheAccessLog.parseLogLine("")
+    }
+  }
+
+  it should "correctly parse all log entry parameters" in {
+    val apacheAccessLog = ApacheAccessLog.parseLogLine(
+      """64.242.88.10 - - [07/Mar/2004:16:20:55 -0800] "GET /twiki/bin/view/Main/DCCAndPostFix HTTP/1.1" 200 5253""")
+    assert(apacheAccessLog.ipAddress == "64.242.88.10")
```

Comment:
[Here](http://www.scalatest.org/user_guide/using_matchers#checkingStrings) is the docs for comparing strings using ScalaTest